api.py

```diff
@@ -29,7 +29,55 @@ def get_algorithms():
 
 def _jsonify(response) -> Response:
     """Pass request data as a response"""
-    return flask.jsonify(response.json()), response.status_code
+    if response.ok:
+        return flask.jsonify(response.json()), response.status_code
+    return "{}", response.status_code
+
+
+CustomResponse = NamedTuple('CustomResponse', [
+    ('json', Callable[[], Dict]),
+    ('status_code', int),
+    ('ok', bool)
+])
+
+Matcher = NamedTuple('Matcher', [
+    ('key', str),
+    ('value', object)
+])
+Selector = Union[Matcher, str, int]
+
+
+def _match_item(matcher: Matcher, collection: Dict) -> bool:
+    return collection[matcher.key] == matcher.value
+
+
+def _matching(matcher: Matcher, collection: List[Dict]) -> List[Dict]:
+    match = partial(_match_item, matcher)
+    return list(filter(match, collection))
+
+
+def _select(data, selectors: List[Selector]):
+    for selector in selectors:
+        if isinstance(selector, Matcher):
+            data = _matching(selector, data)[0]
+        else:
+            data = data[selector]
+    return data
```

Comment:
`_match_item`, `_matching`, `_select` should be covered by unit tests.

Reply:
Accepted.